### PR TITLE
Issue1709 reimplement temp comp

### LINKED ISSUE #1709: Make zone temperature comparison being modularized
Implementing the VAV single zone AHU controller `Buildings.Controls.OBC.ASHRAE.G36_PR1.AHUs.SingleZone.VAV.Controller` from bottom to top would require the instantiation of block `Buildings.Controls.OBC.ASHRAE.G36_PR1.Generic.SetPoints.OperationMode`, which requires vectorized variable to calculate the zone status using block like `MultiSum` and `MultiMin`. 

By making the zone temperature comparison being an independent block and separated from the `Buildings.Controls.OBC.ASHRAE.G36_PR1.Generic.SetPoints.OperationMode`, will avoid vectorized variable for composing the VAV single zone AHU controller.

In addition, the block `Buildings.Controls.OBC.ASHRAE.G36_PR1.TerminalUnits.Reheat.SystemRequests` also implemented zone temperature comparison. 

This issue intends to make the zone temperature being modularized and to reimplement blocks `Buildings.Controls.OBC.ASHRAE.G36_PR1.Generic.SetPoints.OperationMode` and `Buildings.Controls.OBC.ASHRAE.G36_PR1.TerminalUnits.Reheat.SystemRequests`.

### PR DESCRIPTION
**Pull request by JayHuLBL**
- [x] developed new block `Buildings.Controls.OBC.ASHRAE.G36_PR1.Generic.SetPoints.ZoneStatus` for checking zone temperature status: comparing zone temperature with the minimum/maximum temperature of the group which the zone is in, also comparing it with the setpoint temperatures.
- [x] reimplemented `Buildings.Controls.OBC.ASHRAE.G36_PR1.Generic.SetPoints.OperationMode` and `Buildings.Controls.OBC.ASHRAE.G36_PR1.TerminalUnits.ModeAndSetPoints` to avoid vector related calculation.
- [x] removed  `Buildings.Controls.OBC.ASHRAE.G36_PR1.AHUs.SingleZone.VAV.ModeAndSetPoints` as it is replaced by the reimplemented `Buildings.Controls.OBC.ASHRAE.G36_PR1.TerminalUnits.ModeAndSetPoints`.
- [x] updated AHU single zone controller `Buildings.Controls.OBC.ASHRAE.G36_PR1.AHUs.SingleZone.VAV.Controller`.
- [x] updated `Buildings.Examples.VAVReheat.Guideline36` to use the updated block `Buildings.Controls.OBC.ASHRAE.G36_PR1.TerminalUnits.ModeAndSetPoints`.

This closes #1709.

### CODE REVIEW

File: Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/CoolingCoil.mo [hunk 1]
```diff
@@ -75,6 +83,8 @@ equation
     annotation (Line(points={{-58,-20},{-42,-20}}, color={255,127,0}));
   connect(uZonSta, intEqu.u2) annotation (Line(points={{-120,-40},{-50,-40},{
           -50,-28},{-42,-28}}, color={255,127,0}));
+  connect(cooCoiPI.y, switch.u1)
+    annotation (Line(points={{12,80},{60,80},{60,8},{70,8}}, color={0,0,127}));
   annotation (defaultComponentName="cooCoi",
         Icon(coordinateSystem(preserveAspectRatio=false), graphics={
         Rectangle(
```
Comment (mwetter) [inline Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/CoolingCoil.mo]: add revision notes.

File: Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/Economizers/Subsequences/Limits.mo [hunk 2]
```diff
@@ -227,24 +225,23 @@ equation
...
+          -62},{-62,-62}}, color={255,0,255}));
+  connect(intLesEqu.y, and3.u2) annotation (Line(points={{-78,-120},{-74,-120},
+          {-74,-70},{-62,-70}}, color={255,0,255}));
+  connect(intEqu1.y, and3.u3) annotation (Line(points={{-78,-160},{-68,-160},{
+          -68,-78},{-62,-78}}, color={255,0,255}));
+  connect(and3.y, not1.u)
+    annotation (Line(points={{-38,-70},{-22,-70}}, color={255,0,255}));
 annotation (Placement(transformation(extent={{-20,110},{0,130}})),
                 Placement(transformation(extent={{-20,20},{0,40}})),
                 Placement(transformation(extent={{60,90},{80,110}})),
```
Comment (mwetter) [inline Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/Economizers/Subsequences/Limits.mo]: Add revision notes

File: Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/ZoneState.mo [hunk 3]
```diff
@@ -82,11 +87,19 @@ equation
...
+          60,24},{68,24}}, color={255,127,0}));
+  connect(addInt.y, addInt1.u1) annotation (Line(points={{92,30},{100,30},{100,6},
+          {108,6}}, color={255,127,0}));
+  connect(booToIntDea.y, addInt1.u2) annotation (Line(points={{92,-70},{100,-70},
+          {100,-6},{108,-6}}, color={255,127,0}));
+  connect(addInt1.y, yZonSta)
+    annotation (Line(points={{132,0},{160,0}}, color={255,127,0}));
   annotation (
         defaultComponentName="zonSta",
         Icon(coordinateSystem(preserveAspectRatio=false), graphics={
```
Comment (mwetter) [inline Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/ZoneState.mo]: add revision notes.

File: Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/Economizers/Subsequences/Limits.mo (hunk 2 above)
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/Economizers/Subsequences/Limits.mo]: added.

File: Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/CoolingCoil.mo (hunk 1 above)
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/CoolingCoil.mo]: added.

File: Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/ZoneState.mo (hunk 3 above)
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/ZoneState.mo]: added.

**Pull request merged**